src/runtime/virtcontainers/pkg/cgroups/utils.go

```diff
@@ -75,6 +79,67 @@ func IsSystemdCgroup(cgroupPath string) bool {
 	return false
 }
 
+func newProperty(name string, units interface{}) systemdDbus.Property {
+	return systemdDbus.Property{
+		Name:  name,
+		Value: dbus.MakeVariant(units),
+	}
+}
+
+func cgroupHierarchy(path string) (cgroups.Hierarchy, cgroups.Path) {
+	if !IsSystemdCgroup(path) {
+		return cgroups.V1, cgroups.StaticPath(path)
+	} else {
+		slice, unit, _ := getSliceAndUnit(path)
+		cgroupSlicePath, _ := systemd.ExpandSlice(slice)
```

Comment:
These calls suggest `cgroupHierarchy()` should be returning an `error`.

Reply:
Right but i wouldn't be able to use `groupHierarchy()` as input to Load() :P, i think it should always pass as the string format is checked in the `IsSystemdCgroup()`

Reply:
You can do something like this I think:

```diff
diff --git a/src/runtime/virtcontainers/pkg/cgroups/cgroups.go b/src/runtime/virtcontainers/pkg/cgroups/cgroups.go
index a0dfdab5..c9a34183 100644
--- a/src/runtime/virtcontainers/pkg/cgroups/cgroups.go
+++ b/src/runtime/virtcontainers/pkg/cgroups/cgroups.go
@@ -137,7 +137,13 @@ func NewCgroup(path string, resources *specs.LinuxResources) (*Cgroup, error) {
                if err != nil {
                        return nil, err
                }
-               if cgroup, err = cgroups.Load(cgroupHierarchy(path)); err == nil {
+
+               cgHierarchy, cgPath, err := cgroupHierarchy(path)
+               if err != nil {
+                       return nil, err
+               }
+
+               if cgroup, err = cgroups.Load(cgHierarchy, cgPath); err == nil {
                        err = cgroup.Update(resources)
                }
        } else {
```

Reply:
Sure i can, i was tempted by the one-liner :) done